addons/binding/org.openhab.binding.hpprinter/src/main/java/org/openhab/binding/hpprinter/internal/handler/WebPutter.java

```diff
@@ -0,0 +1,116 @@
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Performs web puts to EWS.
+ *
+ * @author Stewart Cossey - Initial contribution
+ */
+@NonNullByDefault
+public class WebPutter {
+
+    private final Logger logger = LoggerFactory.getLogger(HPPrinterHandler.class);
+
+    private final HPPrinterConfiguration config;
+    private final String endpoint;
+
+    WebPutter(String endpoint, HPPrinterConfiguration config) {
+        this.endpoint = endpoint;
+        this.config = config;
+    }
+
+    private String authHeader() {
+        logger.trace("PUT Request using auth");
+        String combined = config.userName + ":" + config.ewsPassword;
+        String basicAuth = "Basic " + new String(Base64.getEncoder().encode(combined.getBytes()));
+        return basicAuth;
+    }
+
+    public boolean put(String outputText) {
+        HttpsURLConnection putSecureRequestConn = null;
+        HttpURLConnection putRequestConn = null;
+        DataOutputStream dataOutputStream = null;
+
+        try {
+            URL url = new URL(config.useSSL ? "https" : "http", config.ipAddress, endpoint);
+
+            if (config.useSSL) {
+                putSecureRequestConn = (HttpsURLConnection) url.openConnection();
+
+                if (config.userName != null && config.ewsPassword != null) {
+                    putSecureRequestConn.setRequestProperty("Authorization", authHeader());
+                }
+                putSecureRequestConn.setRequestMethod("PUT");
+
+                putSecureRequestConn.setRequestProperty("Content-Length", Integer.toString(outputText.length()));
+
+                putSecureRequestConn.setDoInput(true);
+                putSecureRequestConn.setDoOutput(true);
+                dataOutputStream = new DataOutputStream(putSecureRequestConn.getOutputStream());
+                dataOutputStream.writeUTF(outputText);
+
+                logger.debug("PUT request response code {}", putSecureRequestConn.getResponseCode());
+            } else {
+                putRequestConn = (HttpURLConnection) url.openConnection();
+
+                if (config.userName != null && config.ewsPassword != null) {
+                    putRequestConn.setRequestProperty("Authorization", authHeader());
+                }
+                putRequestConn.setRequestMethod("PUT");
+
+                putRequestConn.setRequestProperty("Content-Length", Integer.toString(outputText.length()));
+
+                putRequestConn.setDoInput(true);
+                putRequestConn.setDoOutput(true);
+                dataOutputStream = new DataOutputStream(putRequestConn.getOutputStream());
+                dataOutputStream.writeUTF(outputText);
+
+                logger.debug("PUT request response code {}", putRequestConn.getResponseCode());
+            }
+            return true;
+        } catch (IOException e) {
+            logger.error("Cannot perform put request", e);
```

Comment:
In [E.4 of the coding guidelines](http://docs.openhab.org/developers/development/guidelines.html#e-logging) we listed what kind of logging level should be used when. Error should only be used:

> `error` logging should only be used to inform the user that something is tremendously wrong in his setup, the system cannot function normally anymore, and there is a need for immediate action. 

Please change this to use the correct logging levels and if possible change the ONLINE/OFFLINE status of thing / bridge.

Also check other occurrences of error logging